applicationconfig/client/src/test/java/com/azure/applicationconfig/ConfigurationAsyncClientTest.java

```diff
@@ -564,15 +563,70 @@ public void listWithKeyAndLabel() {
                 .assertNext(response -> assertConfigurationEquals(expected, response))
                 .verifyComplete();
 
-        StepVerifier.create(client.listSettings(new SettingSelector().key(key).label(label)))
+        StepVerifier.create(client.listSettings(new SettingSelector().keys(key).labels(label)))
                 .assertNext(configurationSetting -> assertConfigurationEquals(expected, configurationSetting))
                 .verifyComplete();
 
-        StepVerifier.create(client.listSettings(new SettingSelector().key(key)))
+        StepVerifier.create(client.listSettings(new SettingSelector().keys(key)))
                 .assertNext(configurationSetting -> assertConfigurationEquals(expected, configurationSetting))
                 .verifyComplete();
     }
 
+    /**
+     * Verifies that ConfigurationSettings can be added and that we can fetch those ConfigurationSettings from the
+     * service when filtering by their keys.
+     */
+    @Test
+    public void listWithMultipleKeys() {
+        final String key = sdkContext.randomResourceName(keyPrefix, 16);
+        final String key2 = sdkContext.randomResourceName(keyPrefix, 16);
+        final ConfigurationSetting setting = new ConfigurationSetting().key(key).value(key);
```

Comment:
I would suggest choosing another value rather than reusing the key. (similarly in your other cases). 
It is possible for our service to accidentally map incorrect values to properties. (ie. map the key value to the value of the configuration setting and this test case would miss that.)

Reply:
Changed